### PR TITLE
[core] registration-provider-refactoring

### PR DESCRIPTION
**Pull request by rex-schilasky**
### Description
* registration provider single threaded send logic
* Register(false) calls removed finally

### CODE REVIEW

File: ecal/tests/cpp/util_test/src/util_getclients.cpp [hunk 1]
```diff
@@ -22,15 +22,18 @@
 #include <gtest/gtest.h>
 
 enum {
-  CMN_MONITORING_TIMEOUT_MS   = (5000),
-  CMN_REGISTRATION_REFRESH_MS = (1000 * 2)
+  CMN_MONITORING_TIMEOUT_MS   = (5000 + 100),
```
Comment (KerstinKeller) [inline ecal/tests/cpp/util_test/src/util_getclients.cpp:25]: Should we read the config the timeout here? or later set the timeout via the config objects? 
We could create much shorter timeouts -> much quicker test execution.

Comment (rex-schilasky) [reply ecal/tests/cpp/util_test/src/util_getclients.cpp:25]: Yes, lets do this in a new PR.

File: ecal/core/src/registration/ecal_registration_provider.cpp [hunk 2]
```diff
@@ -84,7 +83,7 @@ namespace eCAL
 
     // start cyclic registration thread
     m_reg_sample_snd_thread = std::make_shared<CCallbackThread>(std::bind(&CRegistrationProvider::RegisterSendThread, this));
-    m_reg_sample_snd_thread->start(std::chrono::milliseconds(Config::GetRegistrationRefreshMs()));
+    m_reg_sample_snd_thread->start(std::chrono::milliseconds(0));
```
Comment (KerstinKeller) [inline ecal/core/src/registration/ecal_registration_provider.cpp:86]: @rex-schilasky Is this supposed to be started with 0 timeout? doesn't seem right!

File: ecal/core/src/registration/ecal_registration_provider.h [hunk 3]
```diff
@@ -50,31 +46,22 @@ namespace eCAL
...
 
-    static std::atomic<bool>                    m_created;
-
-    std::unique_ptr<CRegistrationSender>        m_reg_sender;
-
-    std::shared_ptr<CCallbackThread>            m_reg_sample_snd_thread;
-
-    std::mutex                                  m_sample_list_mtx;
-    Registration::SampleList                    m_sample_list;
+    static std::atomic<bool>             m_created;
```
[the remark this answers is not in the document]
Comment (rex-schilasky) [reply ecal/core/src/registration/ecal_registration_provider.h:56]: false positive

File: ecal/core/src/registration/ecal_registration_provider.cpp (hunk 2 above)
Comment (rex-schilasky) [reply ecal/core/src/registration/ecal_registration_provider.cpp]: extent the CThread implementation to support trigger functionality

**Review by KerstinKeller — APPROVED** [review]

**Pull request merged**